flow/project.py

```diff
@@ -104,20 +104,63 @@
 
 class _condition(object):
 
-    def __init__(self, condition):
+    # All condition functions created using the _condition class are stored in this
+    # global registry variable, allowing direct equivalence comparison between
+    # condition objects.
+    _CONDITION_FUNCTIONS = dict()
+
+    def __init__(self, condition, _internal_lambda=False):
+        """The _internal_lambda argument is used to indicate that a condition
+        function is part of the built-in registry of lambdas and can be
+        identified as such during graph detection."""
```

Comment:
I find the `_internal_lambda` parameter and handling very confusing. It took me a while to understand this and I am still not quite sure that I really do. 

This is really only used in combination with `_get_func()`, correct?

Reply:
Yes, you are correct. The reason I introduced it is because in general we decided not to try and use any sort of code introspection to determine equality of lambda functions, but all of our pre-made conditions are implemented as lambdas. I use this flag later to distinguish between the lambdas we create internally and the lambdas that users pass in.

Reply:
Maybe we should provide a thin-wrapper that users can use to wrap their anonymous functions that is slightly less intrusive:
```
def with_name(name, func):
    setattr(func, '_flow_func_name', name)
    return func

@Project.pre(with_name('myfunc', lambda job: ... ))
# ...
```